Auth/Platform/Auth.Platform.Android/Credentials/OIDCCredentialManager.cs

```diff
@@ -0,0 +1,73 @@
+﻿using System;
+using AeroGear.Mobile.Auth.Credentials;
+using Android.App;
+using Android.Content;
+using Java.Lang;
+
+namespace AeroGear.Mobile.Auth.Credentials
+{
+    /// <summary>
+    /// Class for peristing, retrieving, updating and removing OpenID Connect
+    /// credentials on a device.
+    /// </summary>
+    public class OIDCCredentialManager : ICredentialManager
+    {
+        private static readonly string StoreName = "AeroGear.Mobile.Auth.Credentials";
+
+        private ISharedPreferences SharedPreferences;
+
+        private static OIDCCredentialManager instance;
+
+        /// <summary>
+        /// Instance of the credential manager using the application context.
+        /// </summary>
+        /// <value>Instance of <see cref="OIDCCredentialManager"/></value>
+        public static OIDCCredentialManager Instance
+        {
+            get
+            {
+                if (instance == null)
+                {
+                    instance = new OIDCCredentialManager(Application.Context.ApplicationContext);
+                }
+                return instance;
+            }
+        }
+
+        private OIDCCredentialManager(Context context)
+        {
+            SharedPreferences = context.GetSharedPreferences(StoreName, FileCreationMode.Private);
+        }
+
+        public ICredential Read(string key)
+        {
+            string currentState = SharedPreferences.GetString(key, null);
+            if (currentState == null)
```

Comment:
Could also use the ternary conditional here if you introduce a `credential` variable